File: 128-longest-consecutive-sequence/128-longest-consecutive-sequence.scala
object Solution {
    def longestConsecutive(nums: Array[Int]): Int = {
       if (nums.isEmpty) 0
    else {
      val sort = nums.sorted
      val (max, _, cnt) = sort.tail.foldLeft(0, sort(0), 1) { case ((mx, prev, acc), e) =>
        if (Math.abs(e - prev) == 1) (mx, e, acc + 1)
          else if (e - prev == 0) (mx, e, acc)
        else (math.max(mx, acc), e, 1)
      }
      math.max(max, cnt)
    }
    }
    //-1 -1 0 1 3 4 5 6 7 8 9
}

File: 128-longest-consecutive-sequence/128-longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        if(nums == null || nums.length == 0) return 0;
        Set<Integer> set = new HashSet<>();
        for (int num: nums) set.add(num);
        int longest = 1, currNum = 0, currLongest = 0;
        for (int num: nums) {
            if (!set.contains(num - 1)){
                currNum = num;
                currLongest = 0;
            }
            while (set.contains(currNum)) {
                currNum += 1;
                currLongest += 1;
            }
            longest = Math.max(currLongest, longest);
        }
        return longest;
    }
}